src/db/file.rs

```diff
@@ -276,6 +254,46 @@ pub fn add_path_into_database<P: AsRef<Path>>(conn: &Connection,
     file_list_to_json(file_list_with_mimes)
 }
 
+#[cfg(not(windows))]
+fn detect_mime(content: &Vec<u8>, file_path: &Path) -> Result<String> {
+    use std::ffi::OsStr;
+    use magic::{Cookie, flags};
+    let cookie = Cookie::open(flags::MIME_TYPE)?;
+    cookie.load::<&str>(&[])?;
+    let mime = cookie.buffer(&content)?;
+
+    // magic is not specific enough sometimes
+    match mime.as_str() {
```

Comment:
This makes me a little nervous ... I want to look around and see what the mime type is used for before we change it.